app/src/main/java/com/framgia/soundcloud/screen/detail/DetailActivity.java

```diff
@@ -68,4 +81,57 @@ protected void onDestroy() {
         mViewModel.onDestroy();
         super.onDestroy();
     }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_detail, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.download) {
+            if (!mIsDownloadable) {
+                Toast.makeText(getApplicationContext(), Constant.CAN_NOT_DOWNLOAD,
+                        Toast.LENGTH_LONG).show();
+                return true;
+            }
+            if (checkPermission()) {
+                mDownloadTrack.startDownload();
+            } else {
+                requestPermission();
+            }
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    protected boolean checkPermission() {
+        int result = ContextCompat.checkSelfPermission(this,
+                android.Manifest.permission.WRITE_EXTERNAL_STORAGE);
+        return result == PackageManager.PERMISSION_GRANTED ? true : false;
```

Comment:
return ContextCompat.checkSelfPermission(this,android.Manifest.permission.WRITE_EXTERNAL_STORAGE)  == PackageManager.PERMISSION_GRANTED